Analysis/interface/TauTuple.h

```diff
@@ -336,16 +345,28 @@
     /* PAT photons */ \
     PHOTON_VAR(Int_t, index) /* index of the electron */ \
     PHOTON_VAR4(Float_t, pt, eta, phi, energy) /* 4-momentum of the electron */ \
-    PHOTON_VAR2(Float_t, sigmaEtaEta, sigmaIetaIeta) /* shower shape spread */ \
-    PHOTON_VAR4(Float_t, e1x5, e2x5, e3x3, e5x5) /* shower shape signatures */ \
-    PHOTON_VAR3(Float_t, r1x5, r2x5, r9) /* shower shape signatures */ \
-    PHOTON_VAR(Float_t, maxEnergyXtal) \
-    PHOTON_VAR4(Float_t, hcalDepth1OverEcal, hcalDepth2OverEcal, hcalDepth1OverEcalBc, hcalDepth2OverEcalBc) /* HCAL over ECAL variables */ \
-    PHOTON_VAR2(Float_t, full5x5_sigmaEtaEta, full5x5_sigmaIetaIeta) /* shower shape spread */ \
-    PHOTON_VAR4(Float_t, full5x5_e1x5, full5x5_e2x5, full5x5_e3x3, full5x5_e5x5) /* shower shape signatures */ \
-    PHOTON_VAR3(Float_t, full5x5_r1x5, full5x5_r2x5, full5x5_r9) /* shower shape signatures */ \
-    PHOTON_VAR(Float_t, full5x5_maxEnergyXtal) \
-    PHOTON_VAR4(Float_t, full5x5_hcalDepth1OverEcal, full5x5_hcalDepth2OverEcal, full5x5_hcalDepth1OverEcalBc, full5x5_hcalDepth2OverEcalBc) /* HCAL over ECAL variables */ \
+    PHOTON_VAR2(Bool_t, passElectronVeto, hasPixelSeed) \
```

Comment:
I am not fully aware either what these variables exactly mean, but yes I can do some research and comment a bit more.

Reply:
Would be great, thanks! While searching for it myself I found this header file [DataFormats/EgammaCandidates/interface/Photon.h](https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/EgammaCandidates/interface/Photon.h), which actually contains way more features. Do you know if they are also accessible through PAT::Photon and it makes sense to add them too?

Reply:
Hi @ofivite, these are in fact what the PHOTON_SHAPE variables in this code are. I added all the variables inside the ShowerShape structure, for both here [1]. On the other side, PHOTON_ variables are the ones from PAT::Photon.

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/EgammaCandidates/interface/Photon.h#L201-L202

Reply:
Yes indeed, but I was more referring to, for example, [`MIPVariables`](https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/EgammaCandidates/interface/Photon.h#L324) or [`IsolationVariables `](https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/EgammaCandidates/interface/Photon.h#L358) and there's other isolation variables there. Or did I misunderstand their meaning and it doesn't make sense (or possible) to include them?

Reply:
- we have all PF candidates in the code, so isolation variables would be duplication (and create unnecessary dependency on photon iso algorithms).
- MIPVariables - I don't know how how they are computed. To be understood if it is safe for us to use them.

Reply:
OK I agree, let's go ahead with the current set of features and then reiterate if needed.
But it would be still nice to have a minimal features description, to avoid garbage in/garbage out path😊